### ISSUE TITLE
Needs to be signed and submitted to Safari Extensions Gallery

### ISSUE THREAD

**Issue by chrisballinger** [hoyois/clicktoplugin#142]
Any takers?

**Comment by swooshswoosh** [comment]
???

**Comment by Diskutant** [comment]
Since El Capitan Safari Extensions can be signed, and all new Extensions in Apples Extension Gallery need to be signed. You can still install non-signed extensions, Safari will ask if you want to trust the extension.

I agree, signing the extension would be nice, and afaik even a free developer account can sign extensions.

**Comment by hoyois** [comment]
> afaik even a free developer account can sign extensions.

There is no free developer account.

**Comment by Diskutant** [comment]
Yes there is: https://developer.apple.com/support/compare-memberships/

But for Safari Extensions the 99$/Year Account is needed. :/

**Comment by hoyois** [comment]
Nope, there isn't. What's listed as "free" there does not allow you to develop anything.

Safari extensions can still be built and distributed with existing certificates, but starting in July 2016 it won't be possible to develop extensions for free. You can still build extensions for yourself but you have to recompile them every time you start Safari.

**Comment by yashendra2797** [comment]
AFAIK You can still use unsigned extensions but they won't update automatically.

**Comment by nabeelr** [comment]
@hoyois I'm willing to submit it to Apple. I have a dev account.

It doesn't violate any apple developer agreements or anything does it?
I use my Apple ID for work, and I can't really risk it being banned or closed or anything like that.

Let me know if you're interested.

**Comment by chrisidefix** [comment]
Some Apple signing certificates appear to have expired in February: https://developer.apple.com/support/certificates/expiration/index.html

Would this also concern the released 3.2 version of ClickToPlugin?

I am asking, because Safari won't let me install the pre-built version on a new system anymore.

**Comment by hoyois** [comment]
@chrisidefix This is a different certificate, you don't even need a developer account to get it, so it's not a problem.

There's a Safari/keychain bug preventing extensions not signed by Apple from being "trusted". A solution is booting in Safe Mode to install such extensions (I've also heard that clicking the "Trust" button with the mouse does the trick, but this needs confirmation).

@nabeelr I'm not the right person to ask about developer agreements. From my perspective, there aren't any real benefits to submitting the extension to the gallery. The issue is that I won't be able to compile the extension after June 2016 (not even for myself!). Are you interested in maintaining CTP after that date?

**Comment by nabeelr** [comment]
@hoyois Sure, I'd be willing to maintain CTP. Mind if I reach out to get some more details from you? nabeel at shopify.com

**Comment by fuzzy76** [comment]
Did anything come out of this? It would be a pity if the extension was completely abandoned.

**Comment by diamondsw** [comment]
If the only concern is the cost of the membership (and not a philosophical disagreement with Apple's terms or requirements), I would flat-out be willing to pay your membership fee. This is simply too important a plugin for the Mac community. Sierra may be ushering in a plugin-free world, but we're not there yet, and even for sites that are HTML5, their players leave something to be desired (we're all looking at you, YouTube).

**Comment by swooshswoosh** [comment]
I would like to make a contribution as well...

**Comment by the-real-tokai** [comment]
When you try to install an 3rd party extension in current Safari (I tested it with 9.x, not sure about 10.x yet) and the Trust dialog appears, try `TAB`+`Space` (aka keyboard control) instead clicking the "Trust" button with the mouse. When clicking with the mouse the dialog just closed for me without the extension being installed.

Also…
Recently I had the funny idea to write my own blocker extension (basically just block some annoying  popup-sites) for myself. Until I learned that you need to pay Apple 99 bucks/year. I was "WTF, Apple?"… it really makes no sense to me to pay 99 bucks/ year just to run my very own stuff on my very own computer. That's insanity!
